modules/ROOT/pages/provisioning-vultr.adoc

```diff
@@ -1,37 +1,64 @@
 = Provisioning Fedora CoreOS on Vultr
 
-This guide shows how to provision new Fedora CoreOS (FCOS) nodes on Vultr.  Fedora CoreOS images are currently not published directly on Vultr, so you must download a Fedora CoreOS Vultr image and upload it to your Vultr account as a https://www.vultr.com/docs/requirements-for-uploading-an-os-iso-to-vultr[custom image].
+This guide shows how to provision new Fedora CoreOS (FCOS) nodes on Vultr. FCOS images are currently not published directly on Vultr, but they can be uploaded as https://www.vultr.com/docs/requirements-for-uploading-an-os-iso-to-vultr[custom images].
 
 == Prerequisites
 
 Before provisioning a FCOS machine, you must have an Ignition configuration file that sets SSH authorized keys for the core user. If you do not have one, see xref:producing-ign.adoc[Producing an Ignition File]. While the Vultr documentation mentions cloud-init and scripts, FCOS does not support cloud-init or the ability to run scripts from user-data. It accepts only Ignition configuration files.
 
+You also need to have access to a Vultr account.  The examples below use the https://github.com/vultr/vultr-cli[vultr-cli] and https://s3tools.org/s3cmd[s3cmd] command-line tools.
 
-You also need to have access to a Vultr account.  The examples below use the https://github.com/vultr/vultr-cli[vultr-cli] command-line tool.
+== Using a custom snapshot
 
-== Creating a Vultr custom snapshot
+Vultr supports creating custom snapshots from public raw images.
 
-Fedora CoreOS is designed to be updated automatically, with different schedules per stream.
+These steps show how to download a FCOS image and upload it to an existing storage bucket, in order to create a snapshot from that.
 
-. Once you have picked the relevant stream, find the corresponding Vultr image on the https://getfedora.org/coreos/download?tab=cloud_operators[download page] and copy the URL of the Download link.
+See https://www.vultr.com/docs/vultr-object-storage[Vultr documentation] for further details on how to create a bucket and configure `s3cmd` to use it.
 
-. Create the custom image:
+=== Creating a snapshot
+
+Fedora CoreOS comes in three streams, with different update schedules per stream. These steps show the `stable` stream as an example, but can be used for other streams too.
+
+. Fetch the latest image suitable for your target stream (or https://getfedora.org/coreos/download/[download and verify] it from the web).
++
+[source, bash]
+----
+STREAM='stable'
+coreos-installer download -s "${STREAM}" -p vultr -f raw.xz --decompress
+----
+
+. Upload the raw image to your bucket, and note its public URL.
```

Comment:
I know you link to https://www.vultr.com/docs/vultr-object-storage earlier in this doc, but I think the user (including me just now) is going to skim over that. Might be worth adding the same or similar information in this step as a `NOTE` or something.

Reply:
Yes, some redundancy doesn't hurt.
Is it fine to tweak this as "Use s3cmd to upload the raw ..." and link that to [Vultr s3cmd example](https://www.vultr.com/docs/vultr-object-storage#s3cmd__Example_CLI_tool)? Or were you missing some specific detail here?

Reply:
I think the part that is slightly confusing to me is that we need a separate set of credentials just for the s3 access, which makes sense, though was hoping there was a way to just to the upload with the vultr CLI and re-use the same creds. The link to the docs helped hammer that point home to me.

Reply:
I was looking for a `vultr-cli` subcommand too, but at this point I think it simply does not exist (and their docs point to s3cmd directly).

Reply:
Amended.